e2e/cephfs.go

```diff
@@ -1962,6 +1962,263 @@ var _ = Describe(cephfsType, func() {
+				if err != nil {
+					framework.Failf("failed to create PVC: %v", err)
+				}
+
+				_, pv, err := getPVCAndPV(f.ClientSet, pvc.Name, pvc.Namespace)
+				if err != nil {
+					framework.Failf("failed to get PV object for %s: %v", pvc.Name, err)
+				}
+
+				originalSnap := getSnapshot(snapshotPath)
+				originalSnap.Namespace = f.UniqueName
+				originalSnap.Spec.Source.PersistentVolumeClaimName = &pvc.Name
+				err = createSnapshot(&originalSnap, deployTimeout)
+				if err != nil {
+					framework.Failf("failed to create snapshot: %v", err)
+				}
+				validateCephFSSnapshotCount(f, 1, subvolumegroup, pv)
+
+				pvcClone, err := loadPVC(pvcClonePath)
+				if err != nil {
+					framework.Failf("failed to load PVC: %v", err)
+				}
+				// Snapshot-backed volumes support read-only access modes only.
+				pvcClone.Spec.AccessModes = []v1.PersistentVolumeAccessMode{v1.ReadOnlyMany}
+				pvcClone.Spec.DataSource.Name = originalSnap.Name
+				pvcClone.Namespace = f.UniqueName
+				err = createPVCAndvalidatePV(f.ClientSet, pvcClone, deployTimeout)
+				if err != nil {
+					framework.Failf("failed to create PVC: %v", err)
+				}
+				_, pvs, err := getPVCAndPV(f.ClientSet, pvcClone.Name, pvcClone.Namespace)
+				if err != nil {
+					framework.Failf("failed to get PV object for %s: %v", pvcClone.Name, err)
+				}
+
+				snap := getSnapshot(snapshotPath)
+				snap.Namespace = f.UniqueName
+				snap.Spec.Source.PersistentVolumeClaimName = &pvcClone.Name
+				for i := 0; i < totalCount; i++ {
+					name := fmt.Sprintf("%s%d", f.UniqueName, i)
+					snap.Name = name
+					err = createSnapshot(&snap, deployTimeout)
+					if err != nil {
+						framework.Failf("failed to create snapshot: %v", err)
+					}
+				}
+				validateCephFSSnapshotCount(f, totalCount, subvolumegroup, pvs)
+				for i := 0; i < totalCount; i++ {
+					name := fmt.Sprintf("%s%d", f.UniqueName, i)
+					snap.Name = name
+					err = deleteSnapshot(&snap, deployTimeout)
+					if err != nil {
+						framework.Failf("failed to delete snapshot: %v", err)
+					}
+				}
+				err = deletePVCAndValidatePV(f.ClientSet, pvcClone, deployTimeout)
+				if err != nil {
+					framework.Failf("failed to delete PVC: %v", err)
+				}
+				err = deleteSnapshot(&originalSnap, deployTimeout)
+				if err != nil {
+					framework.Failf("failed to delete snapshot: %v", err)
+				}
+				validateCephFSSnapshotCount(f, 0, subvolumegroup, pv)
+				err = deletePVCAndValidatePV(f.ClientSet, pvc, deployTimeout)
+				if err != nil {
+					framework.Failf("failed to delete PVC: %v", err)
+				}
+				err = deleteResource(cephFSExamplePath + "storageclass.yaml")
+				if err != nil {
+					framework.Failf("failed to delete CephFS storageclass: %v", err)
+				}
+
+				err = createCephfsStorageClass(f.ClientSet, f, false, nil)
+				if err != nil {
+					framework.Failf("failed to create CephFS storageclass: %v", err)
+				}
+			})
+
+			By("checking clone of shallow snapshot", func() {
```

Comment:
@riya-singhal31 
Is this testcase implementation of the comment @Madhu-1 mentioned [here](https://github.com/ceph/ceph-csi/pull/3717#discussion_r1190708180) ?

Step 3. Create a ROX clone ->  implies restore the snapshot into a ROX PVC.